include/pmacc/memory/boxes/DataBox.hpp

```diff
@@ -27,48 +27,31 @@
 
 namespace pmacc
 {
-    namespace detail
-    {
-        template<typename DataBox>
-        HDINLINE decltype(auto) access(const DataBox& db, DataSpace<1> const& idx = {})
-        {
-            return db[idx.x()];
-        }
-
-        template<typename DataBox>
-        HDINLINE decltype(auto) access(const DataBox& db, DataSpace<2> const& idx = {})
-        {
-            return db[idx.y()][idx.x()];
-        }
-
-        template<typename DataBox>
-        HDINLINE decltype(auto) access(const DataBox& db, DataSpace<3> const& idx = {})
-        {
-            return db[idx.z()][idx.y()][idx.x()];
-        }
-    } // namespace detail
-
     template<typename Base>
     struct DataBox : Base
     {
-        HDINLINE DataBox() = default;
+        DataBox() = default;
 
         HDINLINE DataBox(Base base) : Base{std::move(base)}
         {
         }
 
-        HDINLINE DataBox(DataBox const&) = default;
+        DataBox(DataBox const&) = default;
 
         HDINLINE decltype(auto) operator()(DataSpace<Base::Dim> const& idx = {}) const
         {
-            ///@todo(bgruber): inline and replace this by if constexpr in C++17
-            return detail::access(*this, idx);
+            return Base::operator[](idx);
```

Comment:
The implementation does not differ but once the function signature is `const` and once not. In the const method `Base` is `const` and therefor ethe const implementation there is called. `decltype(auto)` is deriving the return type of the call and provide a const reverence in case we call the const method else a normal reference.